### ISSUE TITLE
Inaccessible generic type name in "nameof" is reported as a method group

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#20444]
**Version Used**: master

**Steps to Reproduce**:

```cs

class C
{
    void M()
    {
        _ = nameof(X<int>.Y<int>); 
    }
}

class X<T>
{
    private void Y<U>()
    {
    }
}
```
**Expected Behavior**:

either of:

`error CS0426: The type name 'Y<>' does not exist in the type 'X<int>'`
`error CS0122: 'X<int>.Y<>' is inaccessible due to its protection level`

**Actual Behavior**:

`error CS8084: Type parameters are not allowed on a method group as an argument to 'nameof'.`

**Comment by alrz** [comment]
In fact if `Y<U>` is missing you would still get the same error.

**Comment by YairHalberstadt** [comment]
Given that this:
```csharp
class C
{
    void M()
    {
        _ = nameof(X.Y<int>); 
    }
}

class X
{
    public void Y<U>()
    {
    }
}
```

generates exactly the same error, I don't see why this is an issue?

**Comment by YairHalberstadt** [comment]
But the message is correct, it just is not the only error?

**Comment by CyrusNajmabadi** [comment]
I don't acctually understand the issue here.  `Type parameters are not allowed on a method group as an argument to 'nameof'.` is the correct error.  You get that with:

```c#
class X<T>
{
    public void Y<U>()
    {
        nameof(Y<int>);
    }
}
```

Accessibility doesn't come into play here with your original cse.  Even if you had:

```
class C
{
    void M()
    {
        _ = nameof(X<int>.Y<int>); 
    }
}

class X<T>
{
    public void Y<U>()
    {
    }
}
```

it wouldn't make this ok.  you still can't have type args.

Finally, if you dropped the args, and the method was inaccessible it would be legal:

```
class C
{
    void M()
    {
        _ = nameof(X<int>.Y); 
    }
}

class X<T>
{
    private void Y<U>()
    {
    }
}
```

So i think this should be closed.
